backend/app/src/main/java/com/moh/phlat/backend/service/SourceDataServiceImpl.java

```diff
@@ -35,81 +36,106 @@ public class SourceDataServiceImpl implements SourceDataService {
 	private SourceDataFilterSpecification specificationService = new SourceDataFilterSpecificationImpl();
 	
 	@Override
-	public List<SourceData> getSourceData(Long controlId, SourceDataFilterParams filterProcess) {
-		
-		Specification<SourceData> combinedSpecification;
+	public List<SourceData> getSourceData(Long controlId, SourceDataFilterParams filterSource, Pageable pageable) {
 		
 		try {
-			combinedSpecification = specificationService.buildSpecificationWhereEqual("controlTableId", controlId.toString());
+			return sourceDataRepository.findAll(buildSpecification(controlId, filterSource), pageable);
 		} catch (Exception e) {
 			logger.error("Error occured: {}", e.getMessage(), e);
 			
 			return new ArrayList();
```

Comment:
returning raw type?

Reply:
Not sure where this is coming from.  Code doesn't have the combinedSpecification in it:
		try {
			return sourceDataRepository.findAll(buildSpecification(controlId, filterSource), pageable);
		} catch (Exception e) {
			logger.error("Error occured: {}", e.getMessage(), e);
			
			return new ArrayList();
		}

Reply:
```suggestion
			return new ArrayList<SourceData>();
```
did not get it. I meant this